This patch addresses 22554075 which has also been filed in Launchpad
under 1538582.

--- cinder-2015.1.2/cinder/volume/drivers/zfssa/zfssaiscsi.py.~2~	2016-01-31 17:10:27.739412163 -0800
+++ cinder-2015.1.2/cinder/volume/drivers/zfssa/zfssaiscsi.py	2016-01-31 17:11:03.123083242 -0800
@@ -1,4 +1,4 @@
-# Copyright (c) 2014, Oracle and/or its affiliates. All rights reserved.
+# Copyright (c) 2014, 2016, Oracle and/or its affiliates. All rights reserved.
 #
 #    Licensed under the Apache License, Version 2.0 (the "License"); you may
 #    not use this file except in compliance with the License. You may obtain
@@ -415,7 +415,7 @@ class ZFSSAISCSIDriver(driver.ISCSIDrive
         iscsi_properties['target_discovered'] = False
         iscsi_properties['target_portal'] = target_portal
         iscsi_properties['target_iqn'] = iqn
-        iscsi_properties['target_lun'] = lun
+        iscsi_properties['target_lun'] = int(lun)
         iscsi_properties['volume_id'] = volume['id']
 
         if 'provider_auth' in provider:
